packages/broker/src/plugins/legacyWebsocket/WebsocketServer.ts

```diff
@@ -87,179 +103,146 @@ export class WebsocketServer extends EventEmitter {
             })
 
         const streams = new StreamStateManager()
-        this.requestHandler = new RequestHandler(streamFetcher, publisher, streams, subscriptionManager, this.metrics, storageNodeRegistry, streamrUrl)
-        networkNode.addMessageListener((msg: Protocol.MessageLayer.StreamMessage) => this._broadcastMessage(msg, streams))
-
-        this._pingInterval = setInterval(() => {
-            this._pingConnections()
-        }, this.pingInterval)
+        this.requestHandler = new RequestHandler(
+            streamFetcher,
+            publisher,
+            streams,
+            subscriptionManager,
+            this.metrics,
+            storageNodeRegistry,
+            streamrUrl
+        )
+        networkNode.addMessageListener((msg: Protocol.MessageLayer.StreamMessage) => this.broadcastMessage(msg, streams))
+
+        this.wss = new WebSocket.Server({
+            server: httpServer,
+            maxPayload: 1024 * 1024
+        })
+        this.wss.on('connection', (ws: WebSocket, request: http.IncomingMessage) => {
```

Comment:
There's an argument to be made of having this as a separate (named) function or private method as it is very long. OTOH it is not called from anywhere else, so I sort of used that as a rationale to keep it here inlined...